### PR TITLE
fix field name of reset password token, add verifier and set status

### PR DESCRIPTION
**Pull request by tomivm**
this PR fixes a bug that allows changing the password of the users of Cboard only with their emails. 
-Don't send the reset password token on /user/forgot response
-Use resBcrypt variable on bcrypt.compare() callback to verify if the token is correct.
-Set the status field of the document to true after changing the password. ( It's better to use this field or delete the document? )

After this merge is necesary modify test of user endpoint because the token is no more available on forgot response. PR #144

### CODE REVIEW

**Review by sylvansson — CHANGES_REQUESTED** [review]
It would be great if you could add some tests directly in this PR to be sure that your changes work as intended. And to answer your question, we can keep the document in order to keep a record of the password reset.

Great work identifying this bug and fixing it :raised_hands:

**Comment by sylvansson** [comment]
@tomivm I think I accidentally pushed to your branch and undid some of your changes. Sorry!

**Comment by sylvansson** [comment]
@tomivm I think I was able to restore your changes.

File: api/controllers/user.js [hunk 1]
```diff
@@ -447,19 +447,18 @@ async function storePassword(req, res) {
       userId: userid,
       status: false,
     }).exec();
-    if (!resetPassword) {
+    const expireTime = moment.utc(resetPassword.resetPasswordExpires).isBefore(moment());
```
Comment (sylvansson) [inline api/controllers/user.js:450]: Will this work properly? If `resetPassword` is undefined then we'll most likely get a TypeError if we try to access `resetPasswordExpires`.

Comment (sylvansson) [inline api/controllers/user.js:450]: I'm not sure it's safe to compare a UTC datetime with a local datetime.

**Review by sylvansson — CHANGES_REQUESTED** [review]

**Comment by sylvansson** [comment]
@tomivm It looks like this PR was closed automatically when I merged #166 because you referenced it with the "resolve" keyword in your other PR. I'm reopening it.

image

File: api/controllers/user.js (hunk 1 above)
Comment (tomivm) [reply api/controllers/user.js:450]: I think you are right! . I delete that if by mistake... I mean the if should persist but with a different message.

Comment (tomivm) [reply api/controllers/user.js:450]: I will try to fix it :)

**Comment by tomivm** [comment]
image
with out UserId and Token:
image
without token:
image
Recived mail:
image
Request and response using the link to change password:
borrar
Changing expired time to 5 seconds and trying to change password:
image
With the actual code on `forgot-password` is only able to forgot one Time the password. what I mean is that if I change My password one Time i cannot request change the password again. when i tried it the response is: 'ERROR: create reset password FAILED '

**Comment by martinbedouret** [comment]
@tomivm I am closing this because it is too old. Could you please check it out and apply the same fix on a new branch in case this is still applicable?

**Pull request closed without merging**